GH-1247 - Upgrade doc updates (#197)
whamcloud/Online-Help: docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md

### Install the Integrated Manager for Lustre Upgrade

The software upgrade process requires super-user privileges to run. Login as the `root` user or use `sudo` to elevate privileges as required.

1. Download the latest Integrated Manager for Lustre release repo:

   ```sh
   yum-config-manager --add-repo=https://github.com/whamcloud/integrated-manager-for-lustre/releases/download/v5.1.0/chroma_support.repo
   ```

1. Verify that the old iml-5.0 repo file has been removed from the repolist and that the 5.1 repo has been added.

   ```bash
   yum repolist
   ```

1. Run the OS upgrade.

   ```bash
   yum clean metadata
   yum update
   ```

Refer to the operating system documentation for details on the correct procedure for upgrading between minor OS releases. Note that a mapping of new installs and upgrades will be displayed. Look through this chart carefully and verify that python2-iml-manager is marked for upgrade and that it will be upgraded to {{site.version}}.

1. Run `chroma-config setup` to complete the installation.

1. Update the repos on each server node (run from the manager node):

   ```bash

>    iml update_repo --hosts mds[1,2],oss[1,2]

Reviewer: This is probably too specific. We should at least have a note that this an example and not canonical.

Author: updated to:
"Update the repos on each server node. As an example, consider the following hosts: mds1.local, mds2.local, oss1.local, and oss2.local:"

   /etc/corosync/corosync.conf \
   /etc/ssh \
   /root/.ssh \
   | (cd $BCKROOT && tar xf -)

   # Pacemaker Configuration:
   cibadmin --query > $BCKROOT/cluster-cfg-$HOSTNAME.xml

   cd `dirname $BCKROOT`
   tar zcf $BCKROOT.tgz `basename $BCKROOT`
   ```

   **Note:** This is not intended to be a comprehensive backup of the entire operating system configuration. It covers the essential components pertinent to Lustre servers managed by Integrated Manager for Lustre that are difficult to re-create if deleted. Make sure to backup any other important configuration files that may be on your system, such as multipath configurations.

   The following files will need to be backed up if multipath is being used on the system:

   ```sh
   /etc/multipath/*
   /etc/multipath.conf
   ```

1. Copy the backups for each server's configuration to a safe location that is not on the servers being upgraded.

## Stopping the filesystem

IML requires that the filesystem(s) associated with each node to be upgraded must be stopped. Follow these steps:

1. Navigate to _Configuration->Filesystems_
1. For each filesystem listed:

   1. Click the filesystem's `Actions` button

>    1. Select _Stop_

Reviewer: We should stop the fs prior to updating the manager.

Author: I moved this section just before the update manager step.

## Upgrade the OS on each server node

> In order to upgrade, make sure yum is configured on each server node to pull down CentOS 7.6 packages. Next, from the manager node, upgrade the OS for each host:

Reviewer: ```suggestion
In order to upgrade, make sure yum is configured on each server node to pull down CentOS 7.7 packages. Next, from the manager node, upgrade the OS for each host:
```

   /var/lib/chroma \
   /etc/sysconfig/network \
   /etc/sysconfig/network-scripts/ifcfg-* \
   /etc/yum.conf \
   /etc/yum.repos.d \
   /etc/hosts \
   /etc/passwd \
   /etc/group \
   /etc/shadow \
   /etc/gshadow \
   /etc/sudoers \
   /etc/resolv.conf \
   /etc/nsswitch.conf \
   /etc/rsyslog.conf \
   /etc/ntp.conf \
   /etc/selinux/config \
   /etc/modprobe.d/iml_lnet_module_parameters.conf \
   /etc/corosync/corosync.conf \
   /etc/ssh \
   /root/.ssh \
   | (cd $BCKROOT && tar xf -)

   # Pacemaker Configuration:
   cibadmin --query > $BCKROOT/cluster-cfg-$HOSTNAME.xml

   cd `dirname $BCKROOT`
   tar zcf $BCKROOT.tgz `basename $BCKROOT`
   ```

   **Note:** This is not intended to be a comprehensive backup of the entire operating system configuration. It covers the essential components pertinent to Lustre servers managed by Integrated Manager for Lustre that are difficult to re-create if deleted. Make sure to backup any other important configuration files that may be on your system, such as multipath configurations.

   The following files will need to be backed up if multipath is being used on the system:

   ```sh
   /etc/multipath/*
   /etc/multipath.conf
   ```

> 1. Copy the backups for each server's configuration to a safe location tha

Author: ```suggestion
1. Copy the backups for each server's configuration to a safe location that is not on the server being upgraded.
```

# Upgrading Integrated Manager for Lustre 5.0.0 to Lustre {{site.lustre_version}} and Integrated Manager for Lustre {{site.version}}

[**Upgrade Guide**](ug_TOC.md)

## Upgrade Integrated Manager for Lustre

> The first component in the environment to upgrade is the Integrated Manager for Lustre server and software. The manager server upgrade can be conducted without any impact to the Lustre file system services.

Reviewer: ```suggestion
The first component in the environment to upgrade is the Integrated Manager for Lustre server and software. The manager server upgrade can be conducted without any impact to the Lustre filesystem services.
```

/etc/ntp.conf \
/etc/selinux/config \
/etc/ssh \
/root/.ssh \
| (cd $BCKROOT && tar xf -)

# IML Database
su - postgres -c "/usr/bin/pg_dumpall --clean" | /bin/gzip > $BCKROOT/pgbackup-`date +\%Y-\%m-\%d-\%H:\%M:\%S`.sql.gz

cd `dirname $BCKROOT`
tar zcf $BCKROOT.tgz `basename $BCKROOT`
```

Copy the backup tarball to a safe location that is not on the server being upgraded.

**Note:** This script is not intended to provide a comprehensive backup of the entire operating system configuration. It covers the essential components pertinent to Lustre servers managed by Integrated Manager for Lustre that are difficult to re-create if deleted.

### Install the Integrated Manager for Lustre Upgrade

The software upgrade process requires super-user privileges to run. Login as the `root` user or use `sudo` to elevate privileges as required.

1. Download the latest Integrated Manager for Lustre release repo:

   ```sh
   yum-config-manager --add-repo=https://github.com/whamcloud/integrated-manager-for-lustre/releases/download/v5.1.0/chroma_support.repo
   ```

1. Verify that the old iml-5.0 repo file has been removed from the repolist and that the 5.1 repo has been added.

   ```bash

>    yum clean all

Reviewer: ```suggestion
```

   yum repolist
   ```

1. Run the OS upgrade.

   ```bash

>    yum clean all

Reviewer: ```suggestion
   yum clean metadata
```

   yum update
   ```

Refer to the operating system documentation for details on the correct procedure for upgrading between minor OS releases. Note that a mapping of new installs and upgrades will be displayed. Look through this chart carefully and verify that python2-iml-manager is marked for upgrade and that it will be upgraded to {{site.version}}.

1. Run `chroma-config setup` to complete the installation.

1. Update the repos on each server node (run from the manager node):

   ```bash

>    iml update-repo <node name>

Reviewer: This is going to be using hostlist expression for the nodes.

Author: Updated to:
```
iml update-repo mds[1,2],oss[1,2]
```

   The following files will need to be backed up if multipath is being used on the system:

   ```sh
   /etc/multipath/*
   /etc/multipath.conf
   ```

1. Copy the backups for each server's configuration to a safe location that is not on the servers being upgraded.

## Stopping the filesystem

IML requires that the filesystem(s) associated with each node to be upgraded must be stopped. Follow these steps:

1. Navigate to _Configuration->Filesystems_
1. For each filesystem listed:

   1. Click the filesystem's `Actions` button
   1. Select _Stop_

## Upgrade the OS on each server node

In order to upgrade, make sure yum is configured on each server node to pull down CentOS 7.6 packages. You may need to disable the CentOS base repo and enable the vault repos. For example:

```bash
sed -i "s/enabled=[0-1]//g" /etc/yum.repos.d/CentOS-Base.repo
sed -i "s/gpgcheck=\([0-1]\)/gpgcheck=\1\nenabled=0/g" /etc/yum.repos.d/CentOS-Base.repo
yum-config-manager --add-repo=http://vault.centos.org/7.6.1810/os/x86_64/
yum-config-manager --add-repo=http://vault.centos.org/7.6.1810/extras/x86_64/

> yum-config-manager --add-repo=http://vault.centos.org/7.6.1810/updates/x86_64/

Reviewer: I think we can skip this. If we target Lustre 2.12.3, it will support EL 7.7

Author: Sure, removed.

```

Next, from the manager node, upgrade the OS for each host using a pdsh expression (you may need to install pdsh via yum).

```bash

>  pdsh -w root@mds[1,2] -w root@oss[1,2] "yum -y upgrade --exclude=python2-iml*"

Reviewer: We should leave out suggestions like pdsh I think and leave that up to the upgraders. Just stick to the upgrade steps directly.

Author: Updated
